src/gui/transferlistsortmodel.h

```diff
@@ -57,5 +59,8 @@ class TransferListSortModel final : public QSortFilterProxyModel
     bool filterAcceptsRow(int sourceRow, const QModelIndex &sourceParent) const override;
     bool matchFilter(int sourceRow, const QModelIndex &sourceParent) const;
 
+    int compare(const QModelIndex &left, const QModelIndex &right) const;
+
     TorrentFilter m_filter;
+    mutable boost::circular_buffer<int> m_sortColumns {boost::circular_buffer<int>::size_type {2}};
```

Comment:
Why stop at `2`, wouldn't other users want more depth?

Reply:
I will argue more than 2 is not practical I don't think anyone can mentally follow such layers of sorting. following https://github.com/qbittorrent/qBittorrent/pull/14423#discussion_r578329656 I will remove the circular buffer altogether.